bin/cron/pardot_sync

```diff
@@ -0,0 +1,419 @@
+	  SELECT email COLLATE utf8_general_ci, name, users.id, 'Teacher', city, state, postal_code, country FROM #{DASHBOARD_DB_NAME}.users AS users
+    LEFT OUTER JOIN #{DASHBOARD_DB_NAME}.user_geos AS user_geos ON user_geos.user_id = users.id
+    WHERE users.user_type = 'teacher'AND LENGTH(email) > 0
+    ON DUPLICATE KEY UPDATE contacts_rollup.email=VALUES(email), contacts_rollup.name=VALUES(name), contacts_rollup.dashboard_user_id=VALUES(dashboard_user_id), roles=VALUES(roles)"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def insert_from_dashboard_pd_enrollments
+  start = Time.now
+  log "Inserting contacts from dashboard.pd_enrollments"
+  PEGASUS_DB.run "
+    INSERT INTO contacts_rollup (email,name)
+	  SELECT email,name FROM dashboard_development.pd_enrollments AS pd_enrollments
+    WHERE LENGTH(pd_enrollments.email) > 0
+	  ON DUPLICATE KEY UPDATE email=contacts_rollup.email,name=contacts_rollup.name"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def insert_from_pegasus_contacts
+  start = Time.now
+  log "Inserting contacts from pegasus.contacts"
+  PEGASUS_DB.run "
+    INSERT INTO contacts_rollup (email, name)
+    SELECT email,name FROM contacts WHERE LENGTH(email) > 0
+    ON DUPLICATE KEY UPDATE contacts_rollup.email=VALUES(email), contacts_rollup.name=VALUES(name)"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def insert_from_pegasus_forms
+  start = Time.now
+  log "Inserting contacts and IP geo data from pegasus.forms"
+  PEGASUS_DB.run "
+    INSERT IGNORE INTO contacts_rollup (email, name, city, state, postal_code, country)
+    SELECT email,name, city, state, postal_code,country FROM forms
+    LEFT OUTER JOIN form_geos on form_geos.form_id = forms.id"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def update_roles
+  start = Time.now
+  log "Updating Facilitator, Workshop Organizer, District Contact roles from dashboard.users"
+  append_to_role_list_from_permission("facilitator", "Facilitator")
+  append_to_role_list_from_permission("workshop_organizer", "Workshop Organizer")
+  append_to_role_list_from_permission("district_contact", "District Contact")
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+
+  start = Time.now
+  log "Updating Professional Learning Partner role"
+  append_plp_to_role_list
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def append_to_role_list_from_permission(permission_name, dest_value)
+  PEGASUS_DB.run "
+    UPDATE contacts_rollup
+	    INNER JOIN #{DASHBOARD_DB_NAME}.users AS users ON users.id = contacts_rollup.dashboard_user_id
+	    INNER JOIN #{DASHBOARD_DB_NAME}.user_permissions AS user_permissions ON user_permissions.user_id = users.id
+	  SET roles = CONCAT(COALESCE(CONCAT(roles,','),''),'#{dest_value}')
+    WHERE LENGTH(users.email) > 0
+		AND user_permissions.permission = '#{permission_name}' AND contacts_rollup.id > 0"
```

Comment:
Given that some UserPermission roles are internal to Code Studio, does it make sense to import all of them into Pardot?